### ISSUE TITLE
Selector timeout manipulation

### ISSUE THREAD

**Issue by rschildmeijer** [rschildmeijer/deft#108]
The intention is that selectorTimeout (in IOLoop) (default 250ms) should be altered to fulfil ioloop timeouts and/or ioloop callbacks 
(i.e. minimize the 
   "time-Timeout-was-executed"  -  "timeouts-timeout"
and/or execute a callback asap.).

Currently the selectorTimeout is always 250ms.

**Comment by rschildmeijer** [comment]
Found this during the impl of #90 and #97. Its a small fix and im working on it.

**Comment by rschildmeijer** [comment]
solution:

```
    long selectorTimeout = 250; // 250 ms
    while (running) {
        try {
            if (selector.select(selectorTimeout) == 0) {
                long ms = tm.execute();
                selectorTimeout = Math.min(ms, 250);
                if (cm.execute()) {
                    selectorTimeout = 1;
                }
                continue;
            }
            .....
```

It's important to have the selectorTimeout > 0. (0 ==  block indefinitely)

**Comment by rschildmeijer** [comment]
Implements #90 (PeriodicCallbacks), #108 (Selector timeout
manipulation), #109 (Asynchronous socket) (closed by c87d5df6fdd379475d6c71b73cdd4cee10bdd246) (closed by c87d5df6fdd379475d6c71b73cdd4cee10bdd246)
(closed by c87d5df6fdd379475d6c71b73cdd4cee10bdd246)
